src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -251,24 +250,52 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   (moduleConfig as any).preloadPlugins = [];
 
   let resourcesLoaded = 0;
-  function setProgress(){
-      resourcesLoaded++;
-      const percentage = resourcesLoaded / totalResources.length * 100;
-      document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
-      document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  function setProgress() {
+    resourcesLoaded++;
+    const percentage = resourcesLoaded / totalResources.length * 100;
+    document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
+    document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  }
+
+  const typesMap: { [key: string]: WebAssemblyBootResourceType | undefined } = {
+    'assembly': 'assembly',
+    'pdb': 'pdb',
+    'icu': 'globalization',
+    'dotnetwasm': 'dotnetwasm',
+  };
+
+  // it would not `loadResource` on types for which there is no typesMap mapping
+  const downloadResource = (request: ResourceRequest): LoadingResource | undefined => {
+    const type = typesMap[request.behavior];
+    if (type !== undefined) {
+      const loaded = resourceLoader.loadResource(request.name, request.resolvedUrl!, request.hash!, type);
+      loaded.url = toAbsoluteUrl(loaded.url);
```

Comment:
Rather than mutating this after the fact, could we change `WebAssemblyResourceLoader` to supply an absolute URL in the first place if that's what's needed?